src/platform/linux/input/inputtino_gamepad.cpp

```diff
@@ -237,12 +245,78 @@ namespace platf::gamepad {
     }
   }
 
-  std::vector<std::string_view> &
-  supported_gamepads() {
-    static std::vector<std::string_view> gps {
-      "auto"sv, "xone"sv, "5"sv, "switch"sv
+  std::bitset<GamepadStatus::GAMEPAD_STATUS>
+  checkGamepadStatus() {
+    std::bitset<GamepadStatus::GAMEPAD_STATUS> status;
+    // Check for uhid device file
+    std::ifstream uhid("/dev/uhid");
+    if (!uhid.good()) {
+      status.set(GamepadStatus::UHID_NOT_AVAILABLE);
+    }
```

Comment:
This part is better handled by inputtino; I think it would be better to try to create a virtual device and then pass back (or log) the actual error:

```c++
auto ds5 = inputtino::PS5Joypad::create({ .name = "Sunshine DualSense (virtual) pad", .vendor_id = 0x054C, .product_id = 0x0CE6, .version = 0x8111 });
if(!ds5){
  ds5.getErrorMessage(); // TODO: do something with this?
}
```

It's basically the same code that is used in `alloc()`.

Reply:
@ABeltramo Even better! I don't need to manually reset after or do some cleanup, right? I'll update.

Reply:
Updated.